### DISCUSSION TITLE
[Internal] Troubleshooting Guide traffic

### DISCUSSION THREAD

**Discussion by Marcono1234** [google/gson#2715] [General]
Since Gson version 2.11.0 many exceptions now refer to the [Troubleshooting Guide](https://github.com/google/gson/blob/main/Troubleshooting.md), and I have also occasionally referred to it on Stack Overflow.

It might be interesting to see how many users visit the Troubleshooting Guide, to get some indication how useful it is, or how well the concept of including the link in the exception messages works. One way to find this out would be to have a look at https://github.com/google/gson/graphs/traffic (I don't have access) and then check if the Troubleshooting Guide appears in the "Popular content" section in the bottom right, and if so how many users have opened it.

@eamonnmcmanus, what do you think? Would you mind checking checking the Traffic tab, and maybe sharing the results here as screenshot or table?

**Comment by eamonnmcmanus** [comment]
Sure, I don't think there are secrets here.

![image](https://github.com/user-attachments/assets/594c99d3-b199-4a99-a426-0517eca77bea)

**Reply by Marcono1234** [reply]
Thanks! I was a bit surprised about how many users still have a look at the `examples/android-proguard-example`. Since that example project seems to be quite outdated, I have created #2718 now proposing to remove it.

The User Guide is possibly also a bit outdated and incomplete; I might create an issue proposing some changes to try to improve it.

**Reply by MaikuB** [reply]
> Thanks! I was a bit surprised about how many users still have a look at the examples/android-proguard-example

Thought I would share it might be that my [repository/plugin](https://github.com/MaikuB/flutter_local_notifications) contributing to this. I don't do native Android or Java/Kotlin development but make use of GSON for a Flutter plugin. When I started including GSON a very long time ago, it looked like docs were still referencing the example. It also looked as though the example got updated as issues were found (e.g. https://github.com/google/gson/commit/d3a75cb56937d687a66c8710e40c2da6223db0c3). Based on this, my belieft was it's intended to be a reference. Consequently, the readme for the Flutter plugin also mentions referring to the example in case it gets updated. It's very likely that some of the stats from pub.dev are from those who find my plugin on there

**Reply by Marcono1234** [reply]
Thanks for the information, that is quite interesting!

As mentioned in #2718 nowadays the most relevant information in `android-proguard-example` might be the ProGuard / R8 configuration there, but even that is outdated.

Maybe it would be better if you linked to https://github.com/google/gson/blob/main/gson/src/main/resources/META-INF/proguard/gson.pro instead, however for users of the latest Gson versions (>= 2.11.0) that configuration is included in the Gson JAR and should be recognized automatically by ProGuard / R8.
It requires however that classes have a no-args constructor and fields use `@SerializedName`.
